### DISCUSSION TITLE
用外网ip访问不到的?  400 Bad Request - Invalid Hostname

### DISCUSSION THREAD

**Discussion by yekainew** [dotnet/WatsonWebserver#93] [General]
Bad Request - Invalid Hostname
HTTP Error 400. The request hostname is invalid.
用外网ip访问不到的?

**Comment by jchristn** [comment]
The ```host``` header on the incoming request must match the listener hostname you supply in the constructor.  Otherwise, you can use ```*```, ```+```, or ```0.0.0.0``` as the listener hostname, but this will require you to run with administrative privileges.

**Comment by yekainew** [comment]
我是用win2019 server的，测试了用 *  ,外网可以访问到了。谢谢

**Reply by jchristn** [reply]
Awesome, glad it is working!  Have a nice day!

**Comment by brianedow** [comment]
Why are administrative privileges required?  Is there any workaround?

**Reply by jchristn** [reply]
If you are using a wildcard hostname, it's required by the OS.  Watson relies on the underlying http.sys and this is an OS limitation.  You should be able to listen on anything other than a wildcard without administrative privileges.
